Remotion/Core/Extensions/Utilities/MemberAccessUtility.cs

```diff
@@ -78,7 +78,7 @@ public static object GetFieldOrPropertyValue (object obj, string fieldOrProperty
     public static object GetFieldOrPropertyValue (object obj, string fieldOrPropertyName, BindingFlags bindingFlags)
     {
       ArgumentUtility.CheckNotNull ("obj", obj);
-      MemberInfo fieldOrProperty = GetFieldOrProperty (obj.GetType (), fieldOrPropertyName, bindingFlags, true);
+      MemberInfo fieldOrProperty = GetFieldOrProperty (obj.GetType (), fieldOrPropertyName, bindingFlags, true)!;
       return GetFieldOrPropertyValue (obj, fieldOrProperty);

```

Comment:
`GetFieldOrPropertyValue` should return `object?`, as `.GetValue(...)` does. Please update usages accordingly.

I think the `CompareValues` delegate signature should also be changed to `public delegate bool CompareValues (object? propertyOrFieldValue, object? compareToValue);`